### PR TITLE
supports MindPXv2 borad which is a product from AirMind.

### PR DESCRIPTION
**Pull request by eflyinghi**
Add a new board from AirMind: MindPX-v2.
We have made flight testing for both px4fmu-v2 and mindpx-v2, since the new codes shouldn't make any side effect to official boards.

The log analyze result is at:
px4fmu-v2: http://logs.uaventure.com/view/7JZAjzSa6Cy2UrtQs7axRA
mindpx-v2: http://logs.uaventure.com/view/mgrcy9cR5Q6vqZXUpbA2ET

And the FWs and whole log could be retrieved at: http://pan.baidu.com/s/1mgZhVgG

Thanks.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Thanks, reviewing!

File: ROMFS/px4fmu_common/init.d/rcS [hunk 1]
```diff
@@ -540,6 +555,49 @@ then
...
 
+	#
+	# MAVLink onboard / TELEM2
+	#
+	if ver hwcmp MINDPX_V2
+	then
+	else
+		# XXX We need a better way for runtime eval of shell variables,
+		# but this works for now
+		if param compare SYS_COMPANION 921600
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rcS]: This is the same for Pixhawk and this board - do we really need to have this separately?

File: src/modules/commander/state_machine_helper.cpp [hunk 2]
```diff
@@ -213,11 +213,19 @@ arming_state_transition(struct vehicle_status_s *status,		///< current vehicle s
 						// are measured but are insufficient
 						if (status->condition_power_input_valid && (status->avionics_power_rail_voltage > 0.0f)) {
 							// Check avionics rail voltages
+#ifdef CONFIG_ARCH_BOARD_MINDPX_V2
```
Comment (LorenzMeier) [inline src/modules/commander/state_machine_helper.cpp]: Why is this necessary?

**Comment by LorenzMeier** [comment]
All in all this looks pretty good! I only had a few minor comments.

File: ROMFS/px4fmu_common/init.d/rcS (hunk 1 above)
Comment (eflyinghi) [inline ROMFS/px4fmu_common/init.d/rcS]: No, we don't need to have this separately.  I've removed the codes.  Thanks Lorenz.

File: src/modules/commander/state_machine_helper.cpp (hunk 2 above)
Comment (eflyinghi) [inline src/modules/commander/state_machine_helper.cpp]: 1. MindPX hardware simplified power management circuit so it does not distinguish power source. A voltage divider resistor is used in USB1 port to clamp the current, so the original 4.5V threshold will trigger low voltage alarm (when input from USB connection, it's actually about 4.3v). Thus we need to adjust this threshold to avoid the situation.

2. The same reason as in 1), adding an usbconnected flag as the condition will not work in this situation.

**Comment by LorenzMeier** [comment]
@eflyinghi Last question: The sensor startup seems to be changed quite a bit. Is this all necessary or could we start the MindPX sensors with just a minor diff?

**Comment by eflyinghi** [comment]
Actually maybe I don’t understand you well.  MindPX started mpu6500, lsm303d, l3gd20, hmc5883 sensors, and no order required.  But when I changed the codes, I found that nsh cannot handle -o,&&,elif operations, so I have to use ugly statement.

File: ROMFS/px4fmu_common/init.d/rc.sensors [hunk 3]
```diff
@@ -106,32 +106,52 @@ else
...
-			if hmc5883 -C -I start
+			if l3gd20 start
+			then
+			fi
+
+			# External I2C bus
+			if hmc5883 -C -T -X start
 			then
 			fi
 		else
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rc.sensors]: Please change this instead to:

```
if ver hwcmp MINDPX_V2
then
fi

if ver hwcmp PX4FMU_V2
then
fi

if ver hwcmp PX4FMU_V1
then
fi
```

The current if / else nesting is unreadable. When you do this the diff should also become readable again.

File: ROMFS/px4fmu_common/init.d/rcS [hunk 4]
```diff
@@ -155,12 +155,27 @@ then
...
+				set USE_IO no
+			else
+				set USE_IO yes
+			fi
 		fi
 	else
 		set USE_IO no
 	fi
 
+	# should set to 0.8 for mindpx-v2 borad.
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rcS]: We are deprecating INAV right now, so you will want to invest some effort into testing LPE or EKF2.

**Comment by LorenzMeier** [comment]
Understood - in that case I prefer a different way of restructuring the if / else which is a bit more verbose but much more readable.

File: ROMFS/px4fmu_common/init.d/rcS (hunk 4 above)
Comment (eflyinghi) [inline ROMFS/px4fmu_common/init.d/rcS]: Thanks Lorenz.  We'll investigate the LPE.

File: ROMFS/px4fmu_common/init.d/rc.sensors (hunk 3 above)
Comment (eflyinghi) [inline ROMFS/px4fmu_common/init.d/rc.sensors]: I made this change, please help review.  Thanks.

**Comment by LorenzMeier** [comment]
Superseded by https://github.com/PX4/Firmware/pull/4119
